### PR TITLE
Enable spot bugs, check styles, code coverage report and avoid running tests when publishing

### PR DESCRIPTION
**Pull request by Maninda**
$subject.
Temporarily disabled 2 tests to avoid the following error.
```
{ballerina}TypeCastError {"message":"incompatible types: 'lang.array:ArrayIterator' cannot be cast to 'lang.array:$anonType$_1'"}

		    	at ballerina.mime.1_0_7:getCaseSensitiveHeaderName(natives.bal:669)
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -16,23 +16,19 @@
...
 ext.stdlibTimeVersion = project.stdlibTimeVersion
 ext.stdlibLogVersion = project.stdlibLogVersion
 ext.stdlibOsVersion = project.stdlibOsVersion
-ext.stdlibIoVersion = project.stdlibIoVersion
-ext.puppycrawlCheckstyleVersion = "8.18"
 ext.testngVersion = "6.14.3"
-ext.slf4jVersion = "1.7.30"
 ext.javaMailVersion = "1.6.2"
 ext.greenmailVersion = "1.5.11"
 ext.mimepullVersion= "1.9.11"
```
Comment (ThisaruGuruge) [inline build.gradle:34]: Shall we move these to the `gradle.properteis` file? Then we can completely remove the lines from 25 to 34

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -4,33 +4,43 @@ on: [push, pull_request]
...
-      - name: Set up JDK 11
-        uses: actions/setup-java@v1
-        with:
-          java-version: 11
-      - name: Build with Gradle
-        env:
-          packageUser: ${{ github.actor }}
-          packagePAT: ${{ secrets.GITHUB_TOKEN }}
-        run: ./gradlew.bat build
+        steps:
```
Comment (ThisaruGuruge) [inline .github/workflows/pull-request.yml:36]: `runs-on` and `steps` should be in the same level.

File: .github/workflows/stdlib-workflow-trigger.yml [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-          -H 'Accept: application/vnd.github.v3+json' \
-          -H 'Authorization: token ${{ secrets.BALLERINA_BOT_TOKEN }}' \
-          --data '{
-            "event_type": "dependency-update"
-          }'
+    trigger-workflow:
+        runs-on: ubuntu-latest
+        if: github.repository_owner == 'ballerina-platform'
+        steps:
+            name: Initiating repository event
```
Comment (ThisaruGuruge) [inline .github/workflows/stdlib-workflow-trigger.yml:16]: Missing array element identifier `-`

File: build.gradle (hunk 1 above)
Comment (Maninda) [reply build.gradle]: Fixed.

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (Maninda) [reply .github/workflows/pull-request.yml]: Fixed.

File: .github/workflows/stdlib-workflow-trigger.yml (hunk 3 above)
Comment (Maninda) [reply .github/workflows/stdlib-workflow-trigger.yml]: Fixed.

File: gradle.properties [hunk 4]
```diff
@@ -1,6 +1,17 @@
...
+javaMailVersion=1.6.2
+greenmailVersion=1.5.11
+mimepullVersion=1.9.11
+puppycrawlCheckstyleVersion=8.18
+checkstyleToolVersion=7.8.2
+githubSpotbugsVersion=4.0.5
+githubJohnrengelmanShadowVersion=5.2.0
+underCouchDownloadVersion=4.0.4
+researchgateReleaseVersion=2.8.0
 ballerinaLangVersion=2.0.0-alpha2-SNAPSHOT
```
Comment (ThisaruGuruge) [inline gradle.properties:15]: We have to bump the ballerina lang version and all the stdlib versions.

Comment (Maninda) [reply gradle.properties]: Bumped.

File: email-ballerina/tests/SmtpComplexEmailSend.bal [hunk 5]
```diff
@@ -18,7 +18,7 @@ import ballerina/jballerina.java;
 import ballerina/mime;
 import ballerina/test;
 
-@test:Config {dependsOn: [testSendEmailWithOptions]}
+// @test:Config {dependsOn: [testSendEmailWithOptions]}
```
Comment (ThisaruGuruge) [inline email-ballerina/tests/SmtpComplexEmailSend.bal:21]:
```suggestion
@test:Config {dependsOn: [testSendEmailWithOptions], enable: false}
```

File: email-ballerina/tests/SmtpEmailSendWithOptions.bal [hunk 6]
```diff
@@ -18,7 +18,7 @@ import ballerina/jballerina.java;
 import ballerina/mime;
 import ballerina/test;
 
-@test:Config {dependsOn: [testSendSimpleEmail]}
+// @test:Config {dependsOn: [testSendSimpleEmail]}
```
Comment (ThisaruGuruge) [inline email-ballerina/tests/SmtpEmailSendWithOptions.bal:21]:
```suggestion
@test:Config {dependsOn: [testSendSimpleEmail], enable: false}
```

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**